File: Exemplos SmartPOS/App_IntentDigitalHUB_Java/Elgin_IntentDigitalHUB_SmartPOS_Java-font/app/src/main/java/com/elgin/java_intentdigitalhub_smartpos/IntentDigitalHubService/TERMICA/Commands/ImprimeXMLNFCe.java
package com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.Commands;

import androidx.annotation.NonNull;

import com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand;
import com.google.gson.JsonObject;


public class ImprimeXMLNFCe extends TermicaCommand {
    final private String dados;
    final private int indexcsc;
    final private String csc;
    final private int param;

    public ImprimeXMLNFCe(@NonNull String dados, int indexcsc, @NonNull String csc, int param) {
        super("ImprimeXMLNFCe");
        this.dados = dados;
        this.indexcsc = indexcsc;
        this.csc = csc;
        this.param = param;
    }

    @Override
    protected JsonObject functionParametersJson() {
        JsonObject functionParametersJson = new JsonObject();

        functionParametersJson.addProperty("dados", this.dados);
        functionParametersJson.addProperty("indexcsc", this.indexcsc);
        functionParametersJson.addProperty("csc", csc);
        functionParametersJson.addProperty("param", this.param);

        return functionParametersJson;
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_SmartPOS_Flutter-font/lib/IntentDigitalHubService/TERMICA/Commands/imprime_xml_nfce.dart
import '../termica_command.dart';

class ImprimeXMLNFCe extends TermicaCommand {
  final String _dados;
  final int _indexcsc;
  final String _csc;
  final int _param;

  ImprimeXMLNFCe(this._dados, this._indexcsc, this._csc, this._param)
      : super('ImprimeXMLNFCe');

  @override
  get functionParametersJson => {
        'dados': _dados,
        'indexcsc': _indexcsc,
        'csc': _csc,
        'param': _param,
      };
}
